File: backend/app/schemas/city.py
"""City schemas — Phase 2 CRUD payloads."""

from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class CityBase(BaseModel):
    prefecture: str
    name: str
    jis_code: str | None = None


class CityCreate(CityBase):
    pass


class CityUpdate(BaseModel):
    prefecture: str | None = None
    name: str | None = None
    jis_code: str | None = None


class CityRead(CityBase):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    created_at: datetime
    updated_at: datetime
    deleted_at: datetime | None = None

File: frontend/lib/schemas/city.ts
/**
 * City (市区町村) zod schemas — Phase 3-13.
 * Mirrors backend `app/schemas/city.py` (CityRead/Create/Update).
 */
import { z } from 'zod';

export const CityReadSchema = z.object({
  id: z.string().uuid(),
  prefecture: z.string().min(1),
  name: z.string().min(1),
  jis_code: z.string().nullable().optional(),
  created_at: z.string(),
  updated_at: z.string(),
  deleted_at: z.string().nullable().optional(),
});

export const CityCreateSchema = z.object({
  prefecture: z.string().min(1),
  name: z.string().min(1),
  jis_code: z.string().nullable().optional(),
});

export const CityUpdateSchema = CityCreateSchema.partial();

export type City = z.infer<typeof CityReadSchema>;
export type CityCreate = z.infer<typeof CityCreateSchema>;
export type CityUpdate = z.infer<typeof CityUpdateSchema>;
